### PR TITLE
Refresh ENS

### PR DESCRIPTION
**Pull request by valentinludu**
Main changes:
- Update user ENS on login
- Update user ENS if user wishes. He can go to his list of identities and click refresh

More Details:
- show wallet refresh button icon on identity list
- place refresh tooltip in the right. It wasn't visible because of the other tooltip.
- add the new swr hooks for refreshing ens
- delete hooks/useGetEnsName.tsx. It wasn't used anywhere.
- remove ethers lib from ens methods

### CODE REVIEW

File: charmClient/hooks/blockchain.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
 
-import { useGETImmutable } from './helpers';
+import { useGETImmutable, usePOST } from './helpers';
 
 export function useGetNFT(attrs: Partial<NFTRequest>) {
   const makeRequest = !!(attrs.chainId && attrs.address && attrs.tokenId);
   return useGETImmutable<NFTData | null>(makeRequest ? `/api/nft/get` : null, attrs);
 }
+
+export function useRefreshENSName() {
```
Comment (mattcasey) [inline charmClient/hooks/blockchain.ts:11]: thanks for continuing to update these

**Review by mattcasey — APPROVED** [review]
lots of cleanup, thanks :)

File: pages/api/v1/bounties/index.ts [hunk 2]
```diff
@@ -186,19 +186,23 @@ async function getBounties(req: NextApiRequest, res: NextApiResponse) {
...
             return null;
           }
 
           return {
-            address: walletAddress as string
+            address: walletAddress
           };
         }
 
+        if (!application.walletAddress) {
```
Comment (motechFR) [inline pages/api/v1/bounties/index.ts:200]: One cleanup suggestion, move the if !wallet.address code block to top for an early exit, and the rest becomes an if / else if block

Comment (valentinludu) [reply pages/api/v1/bounties/index.ts:200]: Done

File: pages/api/v1/bounties/index.ts [hunk 3]
```diff
@@ -186,19 +186,23 @@ async function getBounties(req: NextApiRequest, res: NextApiResponse) {
           application.walletAddress.endsWith('.eth') &&
           ethers.utils.isValidName(application.walletAddress)
         ) {
-          const walletAddress = await resolveENSName(application.walletAddress as string);
+          const walletAddress = await resolveENSName(application.walletAddress);
```
[the remark this answers is not in the document]
Comment (valentinludu) [reply pages/api/v1/bounties/index.ts:189]: Done

**Pull request merged**